apps/openmw/mwrender/shadow.hpp

```diff
@@ -0,0 +1,28 @@
+#ifndef OPENMW_MWRENDER_SHADOW_H
+#define OPENMW_MWRENDER_SHADOW_H
+
+#include <osgShadow/ViewDependentShadowMap>
+
+namespace MWRender
+{
+    class MWShadow : public osgShadow::ViewDependentShadowMap
```

Comment:
I suggest this class to go into components/ (sceneutil?) as nothing about it is Morrowind (or mwrender) specific really.

Reply:
Additionally, we might want to just copy the whole file from OSG, instead a subclass with only a few overridden methods. If something upstream changes that our overriddes can't cope with, then we have a problem.

Reply:
Yeah, I'm becoming increasingly aware that it's probably better off in `components`. It might be a reasonable idea to add shadows to the editor, too, after moving it.

Reply:
> Additionally, we might want to just copy the whole file from OSG, instead a subclass with only a few overridden methods. If something upstream changes that our overriddes can't cope with, then we have a problem.

I initially missed this, so sorry for the delayed comment. Are there definitely no licence issues with taking a whole file? OSG isn't GPL, so I don't want to be the cause of an angry Robert Osfield getting upset that we sublicensed without permission.

Reply:
We've already taken a substantial portion of the file, it shouldn't make a difference whether we do just that or the whole file.

It's fine to do this as long as any license/copyright notices in the file are retained and we clarify that there are changes and what they are. I've already done the same for 'osgDB::ObjectCache' when I needed to massage it a bit. In this case I would assume that any modifications are under the OSG license, not GPL. This, along with listing the changes, will be handy if we want to contribute our changes upstream at some point.